File: Sprint1/C++/H/main.cpp
// https://contest.yandex.ru/contest/22449/problems/H/
#include <iostream>

std::string bin_summ(std::string& num1, std::string& num2) {
    std::string result;

    int carry = 0;
    int i = (int) num1.length() - 1;
    int j = (int) num2.length() - 1;

    while (i >= 0 || j >= 0 || carry) {
        int sum = carry;

        if (i >= 0) {
            sum += num1[i] - '0';
            i--;
        }
        if (j >= 0) {
            sum += num2[j] - '0';
            j--;
        }

        carry = sum / 2;
        sum = sum % 2;

        result.push_back('0' + sum);
    }
    std::string result_rev;

    for (int k = (int) result.length() - 1; k >= 0; k--) {
        result_rev.push_back(result[k]);
    }

    return result_rev;
}

int main() {
    std::string num1, num2;

    std::cin>> num1;
    std::cin>> num2;

    std::cout<<bin_summ(num1, num2)<<std::endl;
    return 0;
}

File: Sprint1/H/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/H/

use std::io;

fn bin_summ(num1: &str, num2: &str) -> String {
    let (mut i, mut j) = (num1.len(), num2.len());
    let mut carry = 0;
    let mut result = Vec::new();

    while i > 0 || j > 0 || carry > 0 {
        let mut sum = carry;

        if i > 0 {
            i -= 1;
            sum += num1.chars().nth(i).unwrap() as u32 - '0' as u32;
        }
        if j > 0 {
            j -= 1;
            sum += num2.chars().nth(j).unwrap() as u32 - '0' as u32;
        }

        carry = sum / 2;
        sum %= 2;

        result.push((sum as u8 + b'0') as char);
    }

    result.reverse();
    result.into_iter().collect()
}

fn main() -> io::Result<()> {
    let mut num1 = String::new();
    let mut num2 = String::new();

    io::stdin().read_line(&mut num1)?;
    io::stdin().read_line(&mut num2)?;

    let (num1, num2) = (num1.trim(), num2.trim());

    println!("{}", bin_summ(num1, num2));

    Ok(())
}